Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/SubmissionsController.cs

```diff
@@ -202,18 +199,47 @@ public ActionResult Update(ModelType model)
                         this.ValidateSubmissionContentLength(model, problem);
                         this.ValidateBinarySubmission(model, problem, submissionType);
                     }
-                }
 
-                if (this.ModelState.IsValid)
-                {
-                    var entity = this.GetById(model.Id) as DatabaseModelType;
-                    entity.Processed = false;
-                    entity.Processing = false;
+                    if (this.ModelState.IsValid)
+                    {
+                        if (this.IsSubmissionProcessing(submission))
+                        {
+                            this.TempData[GlobalConstants.DangerMessage] = Resource.Submission_is_processing;
+                            return this.RedirectToAction(nameof(this.Update), "Submissions", new { area = "Administration" });
+                        }
 
-                    this.UpdateAuditInfoValues(model, entity);
-                    this.BaseUpdate(model.GetEntityModel(entity));
-                    this.TempData.AddInfoMessage(Resource.Successful_edit_message);
-                    return this.RedirectToAction(GlobalConstants.Index);
+                        var submissionProblemId = model.ProblemId.Value;
+                        var submissionParticipantId = model.ParticipantId.Value;
+
+                        using (var scope = new TransactionScope())
+                        {
+                            submission.Processed = false;
+                            submission.Processing = false;
+
+                            this.Data.SubmissionsForProcessing.AddOrUpdate(submission.Id);
+                            this.Data.SaveChanges();
+
+                            var submissionIsBestSubmission = this.IsBestSubmission(
+                                submissionProblemId,
+                                submissionParticipantId,
+                                submission.Id);
+
+                            if (submissionIsBestSubmission)
+                            {
+                                this.Data.ParticipantScores.RecalculateParticipantScore(submissionParticipantId, submissionProblemId);
```

Comment:
This line is more than 120 characters. `.RecalculateParticipantScore(submissionParticipantId, submissionProblemId);` on enw line